src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
+        } else {
+            addIfLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size) {
+            addIfLast(value);
+        } else {
+            Node<T> standStill = searchNode(index).prev;
+            Node<T> newNode = new Node<>(standStill, value, searchNode(index));
+            searchNode(index).prev = newNode;
+            if (standStill == null) {
+                first = newNode;
+            } else {
+                standStill.next = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        T[] inputArray = (T[]) list.toArray();
+        int lengthInputArray = inputArray.length;
+        if (lengthInputArray == 0) {
+            return;
+        }
+        Node<T> standStill = last;
+        for (T element : inputArray) {
+            Node<T> newNode = new Node<>(standStill, element, null);
+            if (standStill == null) {
+                first = newNode;
+            } else {
+                standStill.next = newNode;
+            }
+            standStill = newNode;
+        }
+        last = standStill;
+        size += lengthInputArray;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return searchNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        Node<T> foundNode = searchNode(index);
+        T oldValue = foundNode.item;
+        foundNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return unlink(searchNode(index));
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(T value) {
+        if (value == null) {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (node.item == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (value.equals(node.item)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size() == 0;
+    }
+
+    private void addIfFirst(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        first = newNode;
+        last = newNode;
+    }
+
+    private void addIfLast(T value) {
+        Node<T> previousListElement = last;
+        Node<T> newNode = new Node<>(previousListElement, value, null);
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(last, value, null);
```
You don't need `previousListElement` variable